cumulus/parachains/runtimes/testing/penpal/src/xcm_config.rs

```diff
@@ -94,10 +100,34 @@ parameter_types! {
 		PalletInstance(TrustBackedAssetsPalletIndex::get()).into();
 }
 
+pub fn derived_from_here<AccountId>() -> AccountId
+where
+	AccountId: Decode + Eq + Clone,
+{
+	b"Here"
+		.using_encoded(|b| AccountId::decode(&mut TrailingZeroInput::new(b)))
+		.expect("infinite length input; no invalid inputs for type; qed")
+}
+
+/// A [`Location`] consisting of a single `Here` [`Junction`] will be converted to the
+///  here `AccountId`.
+pub struct HereIsPreset<AccountId>(PhantomData<AccountId>);
+impl<AccountId: Decode + Eq + Clone> ConvertLocation<AccountId> for HereIsPreset<AccountId> {
+	fn convert_location(location: &Location) -> Option<AccountId> {
+		if location.contains_parents_only(0) {
+			Some(derived_from_here::<AccountId>())
+		} else {
+			None
+		}
+	}
+}
+
 /// Type for specifying how a `Location` can be converted into an `AccountId`. This is used
 /// when determining ownership of accounts for asset transacting and when attempting to use XCM
 /// `Transact` in order to determine the dispatch Origin.
 pub type LocationToAccountId = (
+	// Here converts to `AccountId`.
+	HereIsPreset<AccountId>,
```

Comment:
What's the purpose of this?

Reply:
To pass the test to use sudo to register token from penpal.

https://github.com/vgeddes/polkadot-sdk/blob/faa2cb15e685ec2e2543cdecd68d6c11e4f8dc49/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2_outbound.rs#L531

```
cargo test -p bridge-hub-westend-integration-tests --lib tests::snowbridge_v2_outbound::register_token_from_penpal 
```

Reply:
https://github.com/paritytech/polkadot-sdk/pull/7402/commits/866222d0ad2b22f94964cecaf46cdf0f1eef654c

Reply:
see https://github.com/paritytech/polkadot-sdk/pull/7402/files#r1986795619